### PR TITLE
Update business_service.go to accept a context.Context

### PR DESCRIPTION
**Pull request by theckman**
This also updates the new methods to not return the `*http.Response`, as it
doesn't seem needed.

Updates #267

### CODE REVIEW

File: business_service.go [hunk 1]
```diff
@@ -76,55 +87,103 @@ func (c *Client) ListBusinessServices(o ListBusinessServiceOptions) (*ListBusine
...
+// CreateBusinessServiceWithContext instead
 func (c *Client) CreateBusinessService(b *BusinessService) (*BusinessService, *http.Response, error) {
-	data := make(map[string]*BusinessService)
-	data["business_service"] = b
-	resp, err := c.post(context.TODO(), "/business_services", data, nil)
+	return c.createBusinessServiceWithContext(context.Background(), b)
+}
+
+// CreateBusinessServiceWithContext creates a new business service.
+func (c *Client) CreateBusinessServiceWithContext(ctx context.Context, b *BusinessService) (*BusinessService, error) {
```
Comment (theckman) [inline business_service.go:104]: @stmcallister similar to #293, I'm curious of your thoughts around removing the `*http.Response` from these methods.

**Comment by theckman** [comment]
This PR also takes naming inspiration from #295.

File: business_service.go [hunk 2]
```diff
@@ -49,12 +49,23 @@ type ListBusinessServiceOptions struct {
...
+	if err != nil {
+		return nil, err
+	}
+
+	return &ListBusinessServicesResponse{BusinessServices: bss}, nil
+}
+
+// ListBusinessServicesPaginated lists existing business services, automatically
+// handling pagination and returning the full collection.
+func (c *Client) ListBusinessServicesPaginated(ctx context.Context, o ListBusinessServiceOptions) ([]*BusinessService, error) {
```
Comment (theckman) [inline business_service.go:62]: This takes naming inspiration from #295.

Comment (stmcallister) [reply business_service.go:67]: Good call.

File: business_service.go (hunk 1 above)
Comment (stmcallister) [reply business_service.go:109]: As mentioned in #293, as long as we can get the same information from somewhere--in this case, from the `APIError` object--we should be okay with this change.

**Review by stmcallister — APPROVED** [review]
Thanks for this! 👍

**Pull request merged**